python/cudf/cudf/_cuda/gpu.pyx

```diff
@@ -20,12 +23,37 @@ from cudf._cuda.gpu cimport underlying_type_attribute as c_attr
 
 class CUDARuntimeError(RuntimeError):
 
-    def __init__(self, status):
+    def __init__(self, cudaError_t status):
         self.status = status
-        cdef bytes name = cudaGetErrorName(<cudaError_t>status)
-        cdef bytes msg = cudaGetErrorString(<cudaError_t>status)
+        cdef str name = cudaGetErrorName(status).decode()
+        cdef str msg = cudaGetErrorString(status).decode()
         super(CUDARuntimeError, self).__init__(
-            '%s: %s' % (name.decode(), msg.decode()))
+            '%s: %s' % (name, msg))
+
+    def __reduce__(self):
+        return (type(self), (self.status,))
+
+
+class CUDADriverError(RuntimeError):
```

Comment:
We should really split these between runtime and driver files for future maintainability / our sanity.

Reply:
Sure. How should would you propose we do that? Also what other things should move?

Reply:
For starters: CUDARuntimeError vs CUDADriverError, CUresult vs cudaError, etc. I think we should differentiate based on header files where we have a corresponding `pxd` for each header file (`cuda.h`, `cuda_runtime_api.h`).

Reply:
We can do this as a follow up. For now this looks good to go.

Reply:
Thanks Keith! Filed as issue ( https://github.com/rapidsai/cudf/issues/4961 ).